drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -18,6 +18,7 @@ class Hmis::Hud::Enrollment < Hmis::Hud::Base
 
   delegate :exit_date, to: :exit, allow_nil: true
 
+  # CAUTION: enrollment.project accessor is overridden below
   belongs_to :project, **hmis_relation(:ProjectID, 'Project'), optional: true
```

Comment:
Can we just replace this with a relation to `project` via the view? That way we can just do `Hmis::Hud::Enrollment.joins(:project)` and it won't filter out WIPs.

It would simplify the `viewable_by` scope below, which can become:

```
  replace_scope :viewable_by, ->(user) do
    return none unless user.permissions?(:can_view_enrollment_details)

    viewable_projects = Hmis::Hud::Project.with_access(user, :can_view_enrollment_details)
    joins(:project).merge(viewable_projects)
  end
```


We may still want the `def project ... ` override below, since it would be faster for a single retrieval. (No need to materialize the view for that).
What do you think?

Reply:
This is somewhat out of scope for this PR, so I'm also happy to defer the work itself. There is a lot code cleanup we can do now that the view is in place, but I can do that after this is merged in

Reply:
> Can we just replace this with a relation to project via the view?
We could, and it would be a nice clean up. But it seems like there's a potential to cause issues around creating/updating the enrollment. For example you couldn't do something like `enrollment.project = another_project`

I tried this and it caused some test failures so left it intact for now. Happy to look into it further in this PR or we could roll this into another issue.

> We may still want the def project ...  override below, since it would be faster for a single retrieval. 
I'd would guess that the current override is slower; when it falls back to using wip it executes three queries instead of one. However for a one-off the performance difference shouldn't be noticeable.

Reply:
Thanks for this insight, linking your comment in https://www.pivotaltracker.com/story/show/185510157